Remotion/Web/Development.WebTesting/WebDriver/Factories/Firefox/FirefoxBrowserFactory.cs

```diff
@@ -50,12 +50,14 @@ public IBrowserSession CreateBrowser (DriverConfiguration driverConfiguration)
       var sessionConfiguration = CreateSessionConfiguration(driverConfiguration);
       var commandTimeout = driverConfiguration.CommandTimeout;
 
+      var extendedFirefoxOptions = _firefoxConfiguration.CreateFirefoxOptions(driverConfiguration);
+
       var firefoxDriverService = GetFirefoxDriverService();
-      var driver = new FirefoxDriver(firefoxDriverService, _firefoxConfiguration.CreateFirefoxOptions(), commandTimeout);
+      var driver = new FirefoxDriver(firefoxDriverService, extendedFirefoxOptions, commandTimeout);
       driver.Manage().Timeouts().AsynchronousJavaScript = driverConfiguration.AsyncJavaScriptTimeout;
       var session = new Coypu.BrowserSession(sessionConfiguration, new CustomSeleniumWebDriver(driver, Browser.Firefox));
 
-      return new FirefoxBrowserSession(session, _firefoxConfiguration, firefoxDriverService.ProcessId);
+      return new FirefoxBrowserSession(session, _firefoxConfiguration, firefoxDriverService.ProcessId, driverConfiguration.Headless);
```

Comment:
Passing the driverConfig (for headless) into the options and then supplying the headless-flag explicitly for session splits the information into two separate parts that only make sense if they are together. is it possible that we need to inject the headless flag into the config / clone+update the config with the current headless flag to make this design clean?

better ideas welcome. 

This may also be why I originally thought of just splitting the cases. If there is no clean way to control the flag during the execution, we may need to initialize just once.